homeassistant/components/salt/device_tracker.py

```diff
@@ -0,0 +1,83 @@
+"""Support for Salt Fiber Box routers."""
+import logging
+
+from saltbox import SaltBox
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import (
+    DOMAIN,
+    PLATFORM_SCHEMA,
+    DeviceScanner,
+)
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_IP = "192.168.1.1"
+DEFAULT_USERNAME = "admin"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_HOST, default=DEFAULT_IP): cv.string,
+        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+def get_scanner(hass, config):
+    """Return the Salt device scanner."""
+    scanner = SaltDeviceScanner(config[DOMAIN])
+    return scanner if scanner.success_init else None
+
+
+class SaltDeviceScanner(DeviceScanner):
+    """This class queries a Salt Fiber Box router."""
+
+    def __init__(self, config):
+        """Initialize the scanner."""
+        host = config[CONF_HOST]
+        username = config[CONF_USERNAME]
+        password = config[CONF_PASSWORD]
+        self.saltbox = SaltBox(f"http://{host}", username, password)
+        self.online_clients = []
+
+        # Test the router is accessible.
+        data = self.get_salt_data()
+        self.success_init = data is not None
```

Comment:
Just do this initial call to get salt data in get_scanner()

Reply:
Moved this check as suggested, thanks for reviewing!